Hashes/MD5.js

```diff
@@ -0,0 +1,221 @@
+//= ===============================================================
+// MD5.js
+//
+// Module that replicates the MD5 Cryptographic Hash
+// function in Javascript.
+//= ===============================================================
+
+// main variables
+const CHAR_SIZE = 8
+
+const S = [
+  7, 12, 17, 22, 7, 12, 17, 22, 7, 12, 17, 22, 7, 12, 17, 22, 5, 9, 14, 20, 5,
+  9, 14, 20, 5, 9, 14, 20, 5, 9, 14, 20, 4, 11, 16, 23, 4, 11, 16, 23, 4, 11,
+  16, 23, 4, 11, 16, 23, 6, 10, 15, 21, 6, 10, 15, 21, 6, 10, 15, 21, 6, 10, 15,
+  21
+]
+
+const K = [
+  0xd76aa478, 0xe8c7b756, 0x242070db, 0xc1bdceee, 0xf57c0faf, 0x4787c62a,
+  0xa8304613, 0xfd469501, 0x698098d8, 0x8b44f7af, 0xffff5bb1, 0x895cd7be,
+  0x6b901122, 0xfd987193, 0xa679438e, 0x49b40821, 0xf61e2562, 0xc040b340,
+  0x265e5a51, 0xe9b6c7aa, 0xd62f105d, 0x02441453, 0xd8a1e681, 0xe7d3fbc8,
+  0x21e1cde6, 0xc33707d6, 0xf4d50d87, 0x455a14ed, 0xa9e3e905, 0xfcefa3f8,
+  0x676f02d9, 0x8d2a4c8a, 0xfffa3942, 0x8771f681, 0x6d9d6122, 0xfde5380c,
+  0xa4beea44, 0x4bdecfa9, 0xf6bb4b60, 0xbebfbc70, 0x289b7ec6, 0xeaa127fa,
+  0xd4ef3085, 0x04881d05, 0xd9d4d039, 0xe6db99e5, 0x1fa27cf8, 0xc4ac5665,
+  0xf4292244, 0x432aff97, 0xab9423a7, 0xfc93a039, 0x655b59c3, 0x8f0ccc92,
+  0xffeff47d, 0x85845dd1, 0x6fa87e4f, 0xfe2ce6e0, 0xa3014314, 0x4e0811a1,
+  0xf7537e82, 0xbd3af235, 0x2ad7d2bb, 0xeb86d391
+]
+
+/**
+ * Converts the string to little endian in groups of 8
+ *
+ * @param {string} str - 32 character string representation (binary)
+ * @returns {string} - little endian string
+ *
+ * @example
+ *      toLittleEndian("11111111111110100011100101000010"); // "01000010001110011111101011111111"
+ */
+function toLittleEndian(str) {
+  return (
+    str.slice(24, 32) + str.slice(16, 24) + str.slice(8, 16) + str.slice(0, 8)
+  )
+}
+
+/**
+ * Converts a positive integer to a hexadecimal string representation
+ *
+ * @param {number} num - A 32 bit positive integer
+ * @returns {string} - hex representation of the number
+ *
+ * @example
+ *      toHex(255); // "ff000000"
+ */
+function toHex(num) {
+  let str = pad(num.toString(16), 8)
+  return str.slice(6, 8) + str.slice(4, 6) + str.slice(2, 4) + str.slice(0, 2)
+}
+
+/**
+ * Adds padding to binary/hex string representation
+ *
+ * @param {string} str - string representation (binary/hex)
+ * @param {int} bits - total number of bits wanted
+ * @return {string} - string representation padding with empty (0) bits
+ *
+ * @example
+ *      pad("10011", 8); // "00010011"
+ */
+function pad(str, bits) {
+  let res = str
+  while (res.length % bits !== 0) {
+    res = '0' + res
```

Comment:
Repeated string concatenation is a theoretically not necessarily efficient (but practically efficient) way to build strings in JavaScript; I don't think it's necessarily good practice to use it here. Just something like `'0'.repeat(res.length % bits) + str` should be cleaner.